### ISSUE TITLE
dom-repeat two-way data binding issue

### ISSUE THREAD

**Issue by pradiprv** [Polymer/polymer#4644]
#### Live Demo
http://jsbin.com/favulo/5/edit?html,output

#### Steps to Reproduce
1. In a demo page, Tap add new button
2. Type something in newly added input
3. Tap add new button again
![add-new0](https://cloud.githubusercontent.com/assets/17213236/26617973/bfa33b4c-45f5-11e7-8600-a85403e7ae89.png)
![add-new1](https://cloud.githubusercontent.com/assets/17213236/26617927/81e288a8-45f5-11e7-8a16-da44c86d3109.png)

#### Expected Results
Newly added input should have blank value

#### Actual Results
Newly added input has pre-filled value
![add-new2](https://cloud.githubusercontent.com/assets/17213236/26617936/90440a8e-45f5-11e7-838a-0457bc836718.png)

### Browsers Affected

- [x] Chrome
- [x] Firefox
- [x] Edge
- [x] Safari 9
- [x] Safari 8
- [x] IE 11

### Versions

- Polymer: v1.9.1

**Comment by masonlouchart** [comment]
There is a funnier test to do! 😃 

Click twice one the _"Add new"_ button, write in the second input.
Click one more time on the _"Add new"_ button and type into the third added input.

Tada! 🎉

**Comment by timeu** [comment]
The problem is your `items` declaration in the `properties` block. 

You need to replace: 
```js
properties: {
  items: {
    type: Array,
    value:  ['John', 'Bob'];
   }
 },
```

```js
properties: {
  items: {
    type: Array,
    value: function() { return ['John', 'Bob'];}
  }
},
```
See the updated fiddle: http://jsbin.com/qaqidadute/edit?html,output

Check also [documentation](https://www.polymer-project.org/1.0/docs/devguide/properties)

> When initializing a property to an object or array value, use a function to ensure that each element gets its own copy of the value, rather than having an object or array shared across all instances of the element.

**Comment by masonlouchart** [comment]
@timeu I agree, the declaration should use a function but I got the same result with your fiddle.

**Comment by timeu** [comment]
@masonlouchart 
You are right. That was not the underlying issue. 
If you replace the array of primitive strings with an array of objects, it seems to work tough: http://jsbin.com/hajowufihi/edit?html,output

I think that is somehow related to the fact that `dom-repeat` does not work well with an array of primitives in `Polymer 1.x` (#1913). This should be fixed in `Polymer 2.x`

**Comment by masonlouchart** [comment]
Nice catch @timeu 👍 

Now @pradiprv has all cards in their hands!

**Comment by kevinpschaaf** [comment]
As @timeu noted in #1913, Polymer 1.x has poor support for mutating primitive array values bound to a dom-repeat.

This has been fixed in Polymer 2.0 by completely changing how array changes are tracked in the system: http://jsbin.com/qacowe/edit?html,output

It is unlikely at this point we'll address the fundamental issues in 1.x leading to these types of issues, so I'm going to close this with recommendations of a.) upgrading to Polymer 2.0, or b.) using the workaround of wrapping primitives in objects.
